### PR TITLE
One of bopm's upstream providers removed their blacklist.

### PR DESCRIPTION
**Pull request by shentino**
This commit removes the dead njabl blacklist from bopm's config file.

Signed-off-by: Raymond Jennings <shentino@gmail.com>
Gentoo-Bug: https://bugs.gentoo.org/473754

### CODE REVIEW

**Comment by a17r** [comment]
Please follow standard commit message format: https://wiki.gentoo.org/wiki/Gentoo_git_workflow#Commit_message_format

It would be a good opportunity to bump EAPI to 6 and fix Copyright.

However: Why exactly patch out commented lines?

**Comment by shentino** [comment]
The commented out lines are meant as an example for the admin to enable by uncommenting as-is.  However it refers to a dead blacklist that is no longer being maintained so it should be removed from the file entirely as it is no longer a valid example.

I'll fix this and open a new request.

**Comment by a17r** [comment]
You don't have to close requests btw, instead you can always force-push to fix your commits afterwards.

Patches should always include a short description about why they were added.

**Comment by shentino** [comment]
oh, thanks!  I'll do that instead then ^^.

You might have guessed that I'm totally brand new to this process.  Where do I add the description for the patch btw, in header lines before the diff section of the patch?

**Comment by a17r** [comment]
Sure. I can recommend the usage of `repoman` for your workflow, it already does some valuable basic checks for your ebuild. Use `git commit --amend` to not pile up multiple commits for the same ebuild. Header lines in the patch file would be the appropriate place, yeah, e.g. if there is a git commit you can directly pick from upstream you get that for free (provided they believe in meaningful commit messages).

**Comment by shentino** [comment]
Unfortunately upstream for bopm is dead (probably why it got pmasked for having a bug and no maintainer before I took over).

I'll get to this later today when RL finishes its own torture tests on me :)

**Comment by shentino** [comment]
How does this one look?

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		-e "s!/some/path/scan.log!/var/log/${PN}/scan.log!" \
+		bopm.conf.sample || die
+
+	# Custom Makefile.am rules do not respect DESTDIR,
+	# thus override sysconfdir and localstatedir.
+	emake \
+		DESTDIR="${D}" \
+		sysconfdir="${ED}"/etc \
+		localstatedir="${ED}"/var/log/bopm \
+		install || die "install failed"
```
Comment (monsieurp) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: You don't know it yet but `emake` dies on its own when failing.

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	newinitd "${FILESDIR}"/bopm.init.d-r1 ${PN}
+	newconfd "${FILESDIR}"/bopm.conf.d-r1 ${PN}
+
+	dodoc ChangeLog INSTALL README TODO
+}
+
+pkg_postinst() {
+	enewuser bopm
+
+	install -d -m 0700 -o bopm -g root "${ROOT}"/var/log/bopm
```
Comment (monsieurp) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: No. This should be dealt with using the Gentoo functions such as `insinto`, `doins`, `fowners` and `fperms`. See https://devmanual.gentoo.org/function-reference/install-functions/.

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	chown bopm "${ROOT}"/etc/bopm.conf
```
Comment (monsieurp) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: Ditto.

**Comment by monsieurp** [comment]
It is not necessary to close the PR.

**Comment by shentino** [comment]
updated

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	sed -i \
+		-e "s!/some/path/bopm.pid!/run/${PN}/${PN}.pid!" \
+		-e "s!/some/path/scan.log!/var/log/${PN}/scan.log!" \
+		bopm.conf.sample || die
+
+	# Custom Makefile.am rules do not respect DESTDIR,
+	# thus override sysconfdir and localstatedir.
+	emake \
+		DESTDIR="${D}" \
+		sysconfdir="${ED}"/etc \
```
Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: should be unnecessary, as `sysconfdir` is already passed to `econf`

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+src_install () {
+	sed -i \
+		-e "s!/some/path/bopm.pid!/run/${PN}/${PN}.pid!" \
+		-e "s!/some/path/scan.log!/var/log/${PN}/scan.log!" \
+		bopm.conf.sample || die
+
+	# Custom Makefile.am rules do not respect DESTDIR,
+	# thus override sysconfdir and localstatedir.
+	emake \
+		DESTDIR="${D}" \
```
Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: always use `${ED}` for consistency

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 6]
```diff
@@ -0,0 +1,63 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit eutils multilib user
```
Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: don't need `multilib` anymore for `get_libdir` in `EAPI=6`

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+SRC_URI="http://static.blitzed.org/www.blitzed.org/${PN}/files/${P}.tar.gz"
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~ppc ~x86"
+IUSE=""
+
+RDEPEND=""
+DEPEND="${RDEPEND}"
+
+DOCS=(ChangeLog INSTALL README TODO)
```
Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: all of these are already in the default `einstalldocs` list, with the exception of `INSTALL`, which imo isn't really necessary. Hence, remove defining `DOCS` completely

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+		DESTDIR="${D}" \
+		sysconfdir="${ED}"/etc \
+		localstatedir="${ED}"/var/log/bopm \
+		install
+
+	fperms 600 /etc/bopm.conf
+
+	# Remove libopm related files, because bopm links statically to it
+	# If anybody wants libopm, please install net-libs/libopm
+	rm -r "${ED}"/usr/$(get_libdir) "${ED}"/usr/include || die
```
Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: Also, replace all mentions of `"${ED}"/` with `"${ED}"` as `$ED` is already slash terminated

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+KEYWORDS="~alpha ~amd64 ~ppc ~x86"
+IUSE=""
+
+RDEPEND=""
+DEPEND="${RDEPEND}"
+
+DOCS=(ChangeLog INSTALL README TODO)
+PATCHES=("${FILESDIR}"/remove-njabl.patch)
+
+src_configure() {
```
Comment (nvinson) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: You'll have to test it, but I think you can get rid of this function completely.

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+src_install () {
+	sed -i \
+		-e "s!/some/path/bopm.pid!/run/${PN}/${PN}.pid!" \
+		-e "s!/some/path/scan.log!/var/log/${PN}/scan.log!" \
+		bopm.conf.sample || die
+
+	# Custom Makefile.am rules do not respect DESTDIR,
+	# thus override sysconfdir and localstatedir.
+	emake \
```
Comment (nvinson) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: patch Makefile.am to honor the DESTDIR variable.  You can do that by creating a separate patch or by using sed to patch the file in the src_prepare() function.

Comment (nvinson) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: remove the emake call afterwards and add 'default' to the beginning of src_install()

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+	rm -r "${ED}"/usr/$(get_libdir) "${ED}"/usr/include || die
+
+	newinitd "${FILESDIR}"/bopm.init.d-r1 ${PN}
+	newconfd "${FILESDIR}"/bopm.conf.d-r1 ${PN}
+
+	einstalldocs
+}
+
+pkg_postinst() {
+	enewuser bopm
```
Comment (nvinson) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: double-check with the dev manual, but I believe everything in this function should actually be at the end of src_install().

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 4 above)
Comment (shentino) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: check

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 5 above)
Comment (shentino) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: check

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 6 above)
Comment (shentino) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: check

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 7 above)
Comment (shentino) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: check

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 8 above)
Comment (shentino) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: check

**Comment by idella** [comment]
another session of changes to come

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 10 above)
Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: I don't think it's about honouring `DESTDIR`, but rather because he wants to specify `$localstatedir`. @shentino try just dropping the whole `emake` call and replacing it with `default`, as `localstatedir` has already been passed via `econf` and should already be in the Makefile. Test whether the resulting layout is the same (which is my guess).

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 11 above)
Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: @nvinson no, as `src_install()` isn't called for binary packages, meaning user creation has to happen in `pkg_*` functions. If the user isn't required during install, I prefer doing it in `pkg_preinst` or `pkg_postinst`, hence it's fine how it's done.

**Comment by SoapGentoo** [comment]
@shentino Finally, after the many rounds of edits, please squash your changes into one logical commit. Learn how git rebase is your friend: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 11 above)
Comment (nvinson) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: > If the user isn't required during install, I prefer doing it in pkg_preinst or pkg_postinst, hence it's fine how it's done.

Performing those actions in pkg_preinst() or in pkg_postinst() is not proper as those actions should not be done in sandbox-unsafe phases.  If binary installation is a concern, the correct phase is pkg_setup().

Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: Binary installation is **always** a concern, because all packages should be installable from pre-built binary packages. The sandbox is never active for `pkg_*` functions, and whether you do it in `pkg_setup` or `pkg_preinst` makes little difference if the user is not required during `src_*` functions. If it's not required, I prefer doing it in `pkg_preinst`, as that allows me to test up to `src_install` and see that the fs layout is correct without it creating unnecessary users.

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 10 above)
Comment (nvinson) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: > I don't think it's about honouring DESTDIR

It most definitely is, if DESTDIR is not honored then removing the emake call in preference to calling the default action will not work as __eapi6_src_install() calls emake DESTDIR="${D}" install.  Therefore, the custom install rules in Makefile.am need to be fixed.

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 11 above)
Comment (nvinson) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: > The sandbox is never active for pkg_* functions

From the [developer manual](https://devmanual.gentoo.org/ebuild-writing/users-and-groups/index.html),
```
Regardless of whether you are adding a group or a user, this should be performed
in the pkg_setup function: pkg_setup is sandbox-safe, ...
```

If this is incorrect, I request that you file a bug in bugs.gentoo.org, so that the issue can be addressed as it's clearly a bug in either the manual or portage.

Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: Also from the same page,

```
You may also use the pkg_preinst or pkg_postinst functions for user creation,
if the user is not required during or before src_install.
```

https://devmanual.gentoo.org/ebuild-writing/functions/pkg_setup/ says that the sandbox is disabled. I'll check this with the portage guys...

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+IUSE=""
+
+RDEPEND=""
+DEPEND="${RDEPEND}"
+
+PATCHES=("${FILESDIR}"/remove-njabl.patch)
+
+src_configure() {
+	econf \
+		--datadir="${EPREFIX}"/usr/share/doc/${PF} \
```
Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: remove the `--datadir` call, as the folder is wrong anyways. The default provided by `econf` (namely, `${EPREFIX}/usr/share`) should suffice.

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 11 above)
Comment (SoapZA) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: Ok, confirmed, `pkg_setup` is sandbox-**DISABLED**, hence I think retaining the user stuff in `pkg_postinst` should be preferred. I don't have any hard feelings though about this.

File: net-misc/bopm/bopm-3.1.3-r2.ebuild [hunk 13]
```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit eutils user
+
+DESCRIPTION="Blitzed Open Proxy Monitor"
+HOMEPAGE="http://www.blitzed.org/bopm/"
```
Comment (SoapGentoo) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: also, please check the `HOMEPAGE`, it seems the site changed too.

Comment (shentino) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: check

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 12 above)
Comment (shentino) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: bopm actually does something weird here, so instead I documented it with a comment

File: net-misc/bopm/bopm-3.1.3-r2.ebuild (hunk 11 above)
Comment (shentino) [inline net-misc/bopm/bopm-3.1.3-r2.ebuild]: devmanual advises putting enewuser in pkg_setup(), and that is required in this case as src_install() chown's some config files to the new uid.

I moved the fowners and fperms calls to src_install to respect the sandbox...and also because I started getting errors after I eapi-bumped.
